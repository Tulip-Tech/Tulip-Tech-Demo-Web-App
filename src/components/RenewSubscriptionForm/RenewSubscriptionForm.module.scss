@use 'src/styles/variables';
@use 'src/styles/theme';

.title {
  margin-bottom: 16px;
  font-weight: var(--body-font-weight-bold);
  font-size: 24px;
}

.paragraph {
  // empty
}

.infoBox {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: variables.$base-spacing;
  padding: calc(#{variables.$base-spacing} / 2) variables.$base-spacing;

  font-size: 14px;
  line-height: 18px;
  text-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 3px 4px rgba(0, 0, 0, 0.12), 0 1px 5px rgba(0, 0, 0, 0.2);
  background: theme.$panel-bg;
  border-radius: 4px;

  > strong {
    line-height: 16px;
    letter-spacing: 0.25px;
  }
}

.price {
  font-size: 14px;
  line-height: 18px;

  > strong {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
    line-height: 26px;
  }
}

.confirmButton {
  margin-bottom: 8px;
}
