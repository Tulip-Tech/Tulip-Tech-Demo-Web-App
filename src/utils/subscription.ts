import { formatPrice } from '#src/utils/formatting';
import type { Offer, Order } from '#types/checkout';

export const getOfferPrice = (offer: Offer) => formatPrice(offer.customerPriceInclTax, offer.customerCurrency, offer.customerCountry);

export const isSVODOffer = (offer: Offer | Order) => offer.offerId?.[0] === 'S';
