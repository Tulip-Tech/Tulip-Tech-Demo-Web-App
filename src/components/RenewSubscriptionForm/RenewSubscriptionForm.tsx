import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './RenewSubscriptionForm.module.scss';

import Button from '#components/Button/Button';
import { formatDate, formatPrice } from '#src/utils/formatting';
import FormFeedback from '#components/FormFeedback/FormFeedback';
import type { Customer } from '#types/account';
import type { Subscription } from '#types/subscription';

type Props = {
  subscription: Subscription;
  customer: Customer;
  error: string | null;
  submitting: boolean;
  onConfirm: () => void;
  onClose: () => void;
};

const RenewSubscriptionForm: React.FC<Props> = ({ subscription, customer, error, submitting, onConfirm, onClose }: Props) => {
  const { t } = useTranslation('account');

  return (
    <div>
      {error ? <FormFeedback variant="error">{error}</FormFeedback> : null}
      <h2 className={styles.title}>{t('renew_subscription.renew_your_subscription')}</h2>
      <p className={styles.paragraph}>{t('renew_subscription.explanation')}</p>
      <div className={styles.infoBox}>
        <p>
          <strong>{subscription.offerTitle}</strong> <br />
          {t('renew_subscription.next_billing_date_on', { date: formatDate(subscription.expiresAt) })}
        </p>
        <p className={styles.price}>
          <strong>{formatPrice(subscription.nextPaymentPrice, subscription.nextPaymentCurrency, customer.country)}</strong>
          <small>/{t(`periods.${subscription.period}`)}</small>
        </p>
      </div>
      <Button
        className={styles.confirmButton}
        color="primary"
        variant="contained"
        label={t('renew_subscription.renew_subscription')}
        onClick={onConfirm}
        fullWidth
        disabled={submitting}
      />
      <Button label={t('renew_subscription.no_thanks')} onClick={onClose} fullWidth />
    </div>
  );
};

export default RenewSubscriptionForm;
