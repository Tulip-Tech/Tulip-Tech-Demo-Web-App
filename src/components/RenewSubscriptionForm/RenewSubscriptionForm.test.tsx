import React from 'react';
import { render } from '@testing-library/react';

import RenewSubscriptionForm from './RenewSubscriptionForm';

import customer from '#test/fixtures/customer.json';
import subscription from '#test/fixtures/subscription.json';
import type { Subscription } from '#types/subscription';
import type { Customer } from '#types/account';

describe('<RenewSubscriptionForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <RenewSubscriptionForm
        customer={customer as Customer}
        subscription={subscription as Subscription}
        onConfirm={vi.fn()}
        onClose={vi.fn()}
        error={null}
        submitting={false}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
