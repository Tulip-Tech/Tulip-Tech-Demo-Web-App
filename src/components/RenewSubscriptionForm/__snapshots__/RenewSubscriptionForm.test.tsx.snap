// Vitest Snapshot v1

exports[`<RenewSubscriptionForm> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="_title_a290fe"
    >
      renew_subscription.renew_your_subscription
    </h2>
    <p>
      renew_subscription.explanation
    </p>
    <div
      class="_infoBox_a290fe"
    >
      <p>
        <strong>
          Annual subscription (recurring)
        </strong>
         
        <br />
        renew_subscription.next_billing_date_on
      </p>
      <p
        class="_price_a290fe"
      >
        <strong>
          € 22,15
        </strong>
        <small>
          /
          periods.year
        </small>
      </p>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _confirmButton_a290fe _primary_f8f296 _fullWidth_f8f296"
    >
      <span>
        renew_subscription.renew_subscription
      </span>
    </button>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        renew_subscription.no_thanks
      </span>
    </button>
  </div>
</div>
`;
